python/eos/figure/figure.py

```diff
@@ -14,125 +14,252 @@
-    visualize a set of statistical samples.
-
-    The description can be either a dictionary (`figure description`) describing a single plot
-    or a list of dictionaries describing a grid of plots.
-
-    +---------------+----------------------------+
-    | `figure description` (dict)                |
-    +===============+============================+
-    | `description` | `plot description` (dict)  |
-    +---------------+----------------------------+
-    | `items`       | list of `item description` |
-    +---------------+----------------------------+
-
-    +-------------+----------------------------------+
-    | `plot description` (dict)                      |
-    +=============+==================================+
-    | `plot_type` | valid plot type (str)            |
-    +-------------+----------------------------------+
-    | ...         | one or several keyword arguments |
-    +-------------+----------------------------------+
-
-    +-------------+----------------------------------+
-    | `item description` (dict)                      |
-    +=============+==================================+
-    | `item_type` | valid item type (str)            |
-    +-------------+----------------------------------+
-    | ...         | one or several keyword arguments |
-    +-------------+----------------------------------+
-
-
-    :param description: A figure description as documented
-    :type description: dict or list of dict
-    :returns: EOS figure
-    :rtype: eos.figure.Figure
+    type:str=field(repr=False, init=False, default='grid')
+    shape:tuple[int, int]
+    plots:list[Plot]
+
+    def __post_init__(self):
+        nrow, ncol = self.shape
+        self._figure, axes = plt.subplots(nrow, ncol, figsize=(3.0 * nrow, 3.0 * ncol))
+        self._axes = axes.flatten('C') # flatten to row-major style
+
+    def draw(self, context:AnalysisFileContext=None, output:str|None=None):
+        context = AnalysisFileContext() if context is None else context
+        for idx, plot in enumerate(self.plots):
+            plot.prepare()
+            plot.draw(self._axes[idx])
+
+        self._figure.tight_layout()
+        if output is not None:
+            self._figure.savefig(output, bbox_inches='tight')
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        _kwargs = _copy.deepcopy(kwargs)
+        _kwargs['plots'] = [PlotFactory.from_dict(**p) for p in kwargs['plots']]
+        return Deserializable.make(cls, **_kwargs)
+
+
+@dataclass(kw_only=True)
+class CornerFigure(Figure):
+    r"""Produces a grid figure representing the variables described by one or several datafiles.
+    Distributions of the variables are shown on the diagonal, while correlations are plotted on the lower-left corner.
+
+    :param contents: List of contents to be drawn. Contents should be dictionnaries containing the path to the data file, a label and optionaly a color.
```

Comment:
Typo in `dictionnaries`, and `optionaly`

Reply:
Resolved.